pymongo/asynchronous/client_bulk.py

```diff
@@ -312,6 +312,9 @@ async def write_command(
                 bwc._fail(request_id, failure, duration)
             # Top-level error will be embedded in ClientBulkWriteException.
             reply = {"error": exc}
+            # Propagate $clusterTime to the caller.
+            if "$clusterTime" in cmd:
+                reply["$clusterTime"] = cmd["$clusterTime"]
```

Comment:
$clusterTime needs to be parsed from the (potential) server error response, not the command. We also need to do the same for operationTime.

This is the method that handles it, it probably makes more sense to call this method at a lower level if we can (perhaps in write_command):
```
client._process_response(command_response, session)
```

Or another alternative.

Reply:
Makes sense, updated.